Content.Server/Power/Components/BreakerComponent.cs

```diff
@@ -7,8 +7,27 @@ namespace Content.Server.Power.Components;
 public sealed class BreakerComponent : Component
 {
     /// <summary>
-    /// Once power supplied exceeds this limit the breaker will pop.
+    /// Once power supplied exceeds this limit the breaker will start to pop.
     /// </summary>
     [DataField("limit", required: true), ViewVariables(VVAccess.ReadWrite)]
     public float Limit;
+
+    /// <summary>
+    /// Time that supply must exceed limit for the breaker to pop.
+    /// </summary>
+    [DataField("popTime"), ViewVariables(VVAccess.ReadWrite)]
+    public TimeSpan PopTime = TimeSpan.FromSeconds(5);
+
+    /// <summary>
+    /// Whether the breaker is starting to pop.
+    /// </summary>
+    [ViewVariables]
+    public bool Overloaded;
+
+    /// <summary>
+    /// When the breaker will pop.
+    /// Set when supply first exceeds limit.
+    /// </summary>
+    [ViewVariables]
+    public TimeSpan NextPop;
```

Comment:
Shouldn't this be a datafield with `TimeOffsetSerializer`?

Reply:
wouldnt that lead to serialization funnies for maps